assets/js/script.js

```diff
@@ -0,0 +1,162 @@
+if timer runs out game over
+- end game -
+produce final score
+*/
+// #########################
+
+document.addEventListener('DOMContentLoaded', function () {
+  var score = 0
+  var timeLeft = 30
+  var noOfRows = 2
+  var ranGrid
+  var isGameOver = false
+  var wholeGrid = document.querySelectorAll('.container')
+  var palette = [
+    ['rgb(20, 153, 105)',
+      'rgb(20, 140, 95)'],
+    ['rgb(235, 152, 20)',
+      'rgb(235, 135, 20)'],
+    ['rgb(255, 237, 0)',
+      'rgb(255, 220, 0)'],
+    ['rgb(50, 150, 255)',
+      'rgb(40, 160, 235)'],
+    ['rgb(40, 40, 40)',
+      'rgb(0, 0, 0)'],
+    ['rgb(160, 255, 0)',
+      'rgb(120, 255, 90)'],
+    ['rgb(252, 76, 76)',
+      'rgb(255, 59, 59)'],
+    ['rgb(255, 105, 208)',
+      'rgb(255, 101, 230)']
+  ]
+  // console.log(palette[3][0])
+
+// initialize the game
+  function init () {
+  // if start button is pressed, game start
+    wholeGrid.innerHTML = ' '
+  // start the timer
+    gameTimer()
+  // put level 1 into empty container
+    gridCreate()
+    isGameOver = false
+  // remove start button and add restart button
+    startButtonTgt.textContent = 'Restart'
+  // disable init() on clicking start button
+    startButtonTgt.removeEventListener('click', init)
+    startButtonTgt.addEventListener('click', function () {
+      location.reload()
+    })
+  }
+
+// check for game over
+  function gameOver () {
+    isGameOver = true
+  // clear the grid if game is over
+    document.querySelector('.container').innerHTML = ' '
+    var endMsg = document.createElement('h3')
+    document.querySelector('.container').appendChild(endMsg)
+    endMsg.textContent = 'Game Over. Thanks for playing!' + 'You scored a ' + score + '/8 in ' + (30 - timeLeft - 1) + ' seconds.'
+
+    // document.querySelector('.container').style.textAlign = 'center'
+  }
+
+// game timer
+  function gameTimer () {
+    // target timer node
+    var timer = document.querySelector('.timer')
+    var timerId = setInterval(countdown, 1000)
+
+    function countdown () {
+      if (timeLeft < 0 || isGameOver === true) {
+        clearTimeout(timerId)
+        gameOver()
+      } else {
+        timer.innerHTML = timeLeft
+        timeLeft--
+      }
+    }
+  }
+
+// ##### grid creator; change level #####
+  function gridCreate () {
+    var gridId = 0
+    for (var i = 0; i < noOfRows; i++) {
+      document.querySelector('.container').innerHTML += '<div class = "row">'
+
+      for (var k = 0; k < noOfRows; k++) {
+        document.querySelector('.container').innerHTML += '<div class = "L' + noOfRows + 'Grid" id = ' + gridId + '>' + '</div>'
+        gridId++
+      }
+      document.querySelector('.container').innerHTML += '</div>'
+    }
+
+    // grid color
+    wholeGrid = document.querySelectorAll('.L' + noOfRows + 'Grid')
+    // console.log(wholeGrid)
+    for (var p = 0; p < noOfRows - 1; p++) {
+      for (var j = 0; j < gridId; j++) {
+        wholeGrid[j].style.backgroundColor = palette[p][0]
```

Comment:
Try to think if each `grid` is an object and has a `color` property. Will it make any difference?